src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionPersister.cs

```diff
@@ -80,6 +80,31 @@ private System.Threading.Tasks.Task OnSettingChangedAsync(object sender, Propert
             return SpecializedTasks.EmptyTask;
         }
 
+        private object GetFirstOrDefaultValue(OptionKey optionKey, IEnumerable<RoamingProfileStorageLocation> roamingSerializations)
+        {
+            // There can be more than 1 roaming location in the order of their priority.
+            // When fetching a value, we iterate all of them until we find the first one that exists.
+            // When persisting a value, we always use the first location.
+            // This functionality exists for breaking changes to persistence of some options. In such a case, there
+            // will be a new location added to the beginning with a new name. When fetching a value, we might find the old
+            // location (and can upgrade the value accordingly) but we only write to the new location so that
+            // we don't interfere with older versions. This will essentially "fork" the user's options at the time of upgrade.
+
+            foreach (var roamingSerialization in roamingSerializations)
+            {
+                var storageKey = roamingSerialization.GetKeyNameForLanguage(optionKey.Language);
+
+                RecordObservedValueToWatchForChanges(optionKey, storageKey);
```

Comment:
I'm not completely certain whether this is the right thing to do. Should only ever listen to changes in the newest location? Or in the one that we found?

Reply:
This seems the most correct - any change to the location we found or any previous ones will affect the result. I'm just not sure whether this will work correctly.

Reply:
Actually, I don't see any problem with this after looking at the implementation.

Reply:
I see why this seemed confusing, but I agree with the conclusion.